Slim/Utils/Scanner/Remote.pm

```diff
@@ -710,25 +709,20 @@ sub parseAACHeader {
 }
 
 sub parseFlacHeader {
-	my ( $http, $track, $args ) = @_;
-	my $header = $http->response->content;
-
-	my $fh = File::Temp->new();
-	$fh->write( $header, length($header) );
-	$fh->seek(0, 0);
-
-	my $info = Audio::Scan->scan_fh( flac => $fh )->{info};
+	my ( $http, $dataref, $track, $args ) = @_;
 
 	Slim::Formats->loadTagFormatForType('flc');
-
+	my $formatClass = Slim::Formats->classForFormat('flc');
+	my $info = $formatClass->parseStream($dataref, $args);
+	
+	return 1 if ref $info ne 'HASH' && $info;
+	
 	$track->initial_block_type( Slim::Schema::RemoteTrack::INITIAL_BLOCK_ALWAYS );
 	$track->audio_initiate( \&Slim::Formats::FLAC::initiateFrameAlign );
+	$track->content_type('flc');
```

Comment:
Is this the change required to render my previous change in Slim::Protocols::HTTP obsolete?

Reply:
yes it is - in fact, the $track->content_type is normally set in Remote.pm, when scanUrl is used (or the parseXXXHeader directly). I forgot to add it in the parseFlacHeader. In case scanUrl/parseXXXHeader is not used, then it seems that it might not be set by any other LMS part (I'm not sure why - my guess it that it should be the duty of the plugin's ProtocolHandler to do that when it bypasses scanUrl) but it has no impact b/c you'll never reach the HTTP.pm code where it is needed as only my new function in Remote.pm set some other $track item that will cause code to go there.